### PR TITLE
Paginate search results in `InstantSearchScreen`

### PR DESCRIPTION
**Pull request by msasikanth**
- Add query to load search by name results as paging source
- Add query to load search by number results as paging source
- Load search results as paging source
- Show patient search results using `PagingItemAdapter`
- When search results are loaded, then show search results
- Add load state listener to search results adapter
- Remove automatic scrolling to top when setting/updating adapter data
- Remove the default adapter set in `onViewCreated`
- Show and hide progress bar when the refresh load state is loading
- Remove `InstantSearchProgressState` from `InstantSearchModel`
- Remove effects to hide and show search no search results view
- Remove `InstantSearchResultsItemType_old`
- Remove `PatientRepository#search`
- Remove search by patient identifier feature flag and make it default
- Remove `PhoneNumberCriteria`
- Update CHANGELOG

### CODE REVIEW

File: app/src/main/java/org/simple/clinic/patient/PatientRepository.kt [hunk 1]
```diff
@@ -68,42 +57,12 @@ class PatientRepository @Inject constructor(
...
     return database
         .patientSearchDao()
         .searchByNamePagingSource(patientName, facilityId)
   }
 
-  private fun searchByPhoneNumber(
-      phoneNumber: String,
-      facilityId: UUID
-  ): List<PatientSearchResult> {
-    return reportTimeTaken(
```
Comment (vinaysshenoy) [inline app/src/main/java/org/simple/clinic/patient/PatientRepository.kt:76]: Can the `reportTimeTaken` method be removed now?

**Review by vinaysshenoy — CHANGES_REQUESTED** [review]
Overall PR looks good, left a couple of comments. Please take a look?

File: app/src/main/java/org/simple/clinic/patient/PatientRepository.kt (hunk 1 above)
Comment (msasikanth) [reply app/src/main/java/org/simple/clinic/patient/PatientRepository.kt:76]: Are we not using that anywhere else apart from `PatinetRepository`? If not, then yeah. We can remove the `reportTimeTaken` method from the project. Will open a different PR for it.

File: app/src/androidTest/java/org/simple/clinic/patient/PatientRepositoryAndroidTest.kt [hunk 2]
```diff
@@ -3135,4 +3135,82 @@ class PatientRepositoryAndroidTest {
...
+    val searchResults = searchResults(query = "Ramesh", facility = currentFacility)
+
+    //then
+    assertThat(searchResults)
+        .containsExactly(
+            "Ramesh Rao",
+            "Mohan Ramesh",
+            "Sai Ramesh"
+        )
+        .inOrder()
```
[the remark this answers is not in the document]
Comment (msasikanth) [reply app/src/androidTest/java/org/simple/clinic/patient/PatientRepositoryAndroidTest.kt]: Yep, the last name in `containsExactly` is from other facility, and above two are from current facility.

File: app/src/main/java/org/simple/clinic/patient/PatientRepository.kt (hunk 1 above)
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/patient/PatientRepository.kt:76]: Yeah, I see in your branch that it is unused. We can remove this and the corresponding method in `Analytics` as well.

**Review by vinaysshenoy — APPROVED** [review]

**Pull request merged**